### PR TITLE
[ADD] Offchain Read support (EIP-3668) for eth_call and ENS resolution

### PR DESCRIPTION
**Pull request by DarthMike**
Add support for offchain read in `eth_call`, and more specifically add support for offchain ENS resolution.

An example to resolve is:
`offchainexample.eth` (both in ropsten and mainnet)

See
https://eips.ethereum.org/EIPS/eip-3668

There's another bit of related but not the same work, but will address this here : #205

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -4,8 +4,8 @@ import PackageDescription
 let package = Package(
     name: "web3.swift",
     platforms: [
-        .iOS(SupportedPlatform.IOSVersion.v11),
-        .macOS(SupportedPlatform.MacOSVersion.v10_12)
+        .iOS(SupportedPlatform.IOSVersion.v14),
+        .macOS(SupportedPlatform.MacOSVersion.v11)
```
Comment (DarthMike) [inline Package.swift:8]: Increased the minimum version to start modernizing APIs; it also allows to use Combine/ and async/await which greatly reduces work for this specific feature with lots of retries etc.

Also preparing min. version to modernize the APIs in the future

File: web3sTests/ENS/ENSTests.swift [hunk 2]
```diff
@@ -24,184 +24,7 @@ class ENSTests: XCTestCase {
...
-    }
-}
-
-
-#if compiler(>=5.5) && canImport(_Concurrency)
-
-@available(macOS 10.15, iOS 13, watchOS 6, tvOS 13, *)
-extension ENSTests {
-    func testGivenRopstenRegistry_WhenExistingDomainName_ResolvesOwnerAddressCorrectly_Async() async {
+    func testGivenRopstenRegistry_WhenExistingDomainName_ResolvesOwnerAddressCorrectly() async {
```
Comment (DarthMike) [inline web3sTests/ENS/ENSTests.swift:27]: Default tests now using async/await

File: web3sTests/OffchainLookup/OffchainLookupTests.swift [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+//
+//  Created by Miguel on 12/05/2022.
+//  Copyright © 2022 Argent Labs Limited. All rights reserved.
+//
+
+@testable import web3
+import XCTest
+import BigInt
+
+struct DummyOffchainENSResolve: ABIFunction {
```
Comment (DarthMike) [inline web3sTests/OffchainLookup/OffchainLookupTests.swift:13]: Our own deployed contract to test revert responses

File: Package.swift (hunk 1 above)
Comment (dnKaratzas) [reply Package.swift:8]: Combine and async/await are available on iOS v13

Comment (DarthMike) [reply Package.swift:8]: Had an issue with `flatMap` versions (somehow Xcode assuming I wanted to use the newer one from 14). Will try fix this if possible and move min. version to 13

Comment (DarthMike) [reply Package.swift]: @dnKaratzas managed to bump version to 13

**Review by mluisbrown — APPROVED** [review]

File: Package.swift (hunk 1 above)
Comment (dnKaratzas) [reply Package.swift]: Perfect ;)

**Pull request merged**